### ISSUE TITLE
Cannot mount image files with more than 8 characters

### ISSUE THREAD

**Issue by PoloniumRain** [schellingb/dosbox-pure#145]
Sorry to bother you again @schellingb lol, but... I can't get DOSBox Pure to mount .BIN, .IMG and .CUE files with more than 8 characters. I know DOS has limitations with file names being longer than 8 characters, but being as DOSBox SVN and DOSBox Core both mount image files with more than 8 characters i assumed this might be a bug with DBP? And also because image files can be mounted with DBP if i shorten the names to 8 characters or less and then edit the .CUE file.

DSP will also display file names incorrectly if they have more than 8 characters. One example: The .CUE and .BIN files for the game Mortal Kombat Trilogy are each named _**TRILOGYPC_FINAL**_ but DSP will instead say _**TRILOG~2.CUE**_ which you can see here: 
![example](https://user-images.githubusercontent.com/50634894/116435434-3f1be500-a843-11eb-89bb-9d5f2f022398.jpg)

This might be why images wont mount, because DBP isn't using their correct names and within the .CUE it specifically refers to _**TRILOGYPC_FINAL.BIN**_ and not something like _**TRILOG~2.BIN**_ so it can't find the file.

I'm on Windows 10 and i've tried RA 1.9.1 and the latest RA nightly.

**Comment by toniosj** [comment]
Does that game use .ogg files ?

**Comment by PoloniumRain** [comment]
Nope, and it happens with any game. I've tried around 9 or 10 games so far. They will all work perfectly fine if i shorten the file names to 8 characters or less and edit the .CUE file. But i'd rather not do that because it might eventually cause problems if a game relies on the original file names.

**Comment by WarlockSVG** [comment]
> Nope, and it happens with any game. I've tried around 9 or 10 games so far. They will all work perfectly fine if i shorten the file names to 8 characters or less and edit the .CUE file. But i'd rather not do that because it might eventually cause problems if a game relies on the original file names.

Try with the 8.3 file name structure "TRILOG~2.BIN" (or whatever you see in the DOSBox PURE stat menu) in the .CUE file and not the lfn.
... ( FILE "TRILOG~2.BIN" BINARY ) and keep the file names "long"
This would be just for "DOSBox Pure"

**Comment by ToniBC** [comment]
I have tried the Absolute Pinball game under the CD name, ABSOLUTE_PINBALL.cue on Retroarch Nighly and Dosbox Pure with no problems. I never fail to read CDs with long names. Surely the fault is that this cd has the wrong cue.

**Comment by PoloniumRain** [comment]
I found the problem...

The image files do not mount if they have more than 8 characters **AND** the game is **NOT** zipped. 

I tried Absolute Pinball being as @ToniBC got it working, and it will work as a zip file with the original long file names for the .CUE and .BIN, but unzip the game and these image files can't be mounted any more.

At first the problem is that when you unzip Absolute Pinball it wont mount because DBP can't find the .CUE file due to the directory structure (the CD/image and game files are in separate folders), but even when i change the structure so that DBP can find all the files, it still wont mount (in the image below you can see the directory structure). DBP will also show the incorrect file name **_ABSOLU~2.CUE._** If i then change the Absolute Pinball .CUE and .BIN names to 8 characters or less and edit the .CUE they will mount correctly. This is the same thing that happens with Mortal Kombat Trilogy and literally any other game that uses a CD. I've tried 11 games that previously wouldn't mount because of the long file names, and they now all work when zipped.

To confirm this i've also tried: 
RA 1.9.0
RA 1.9.1
Another PC with a separate clean install of RA 1.9.2 (both PC's use Win 10)
Latest RA nightly
DBP 0.11
DBP 0.12

I know i could just use zips from now on but i've had issues with zipped games before and it will take ages to zip all my games.


![examples](https://user-images.githubusercontent.com/50634894/116923253-577a6e00-ac4e-11eb-9e49-cf255c808284.png)

**Comment by schellingb** [comment]
Also worth mentioning, the problem only arises with .CUE files that have a file reference to a .BIN file with a long file name.
DOSBox Pure, unlike any other variant of DOSBox, can load disc images from inside the mounted DOS filesystem. That's how mounting out of ZIP files works. And that's where the 8.3 limit comes in.  So CUE/BIN pairs work for long file names from inside ZIP files because there I made the file naming predictable, not using the ~# scheme.

The fix I committed just now should make the core problem go away. Sadly it will still display TRILOG~2.CUE because as I said it's based on enumerating the DOS filesystem. Improving that would take a bit more time especially for mounted ZIP files because there the original file name information isn't available anymore after loading.

**Comment by PoloniumRain** [comment]
> The fix I committed just now should make the core problem go away. Sadly it will still display TRILOG~2.CUE because as I said it's based on enumerating the DOS filesystem. Improving that would take a bit more time especially for mounted ZIP files because there the original file name information isn't available anymore after loading.
> 
> Edit: Fix released in 0.13.

Just tried 0.13 on two PC's and it works great on both. All images can now be mounted. Thanks for another speedy fix!

I also tested your [other recent fix](https://github.com/schellingb/dosbox-pure/issues/143) on another problem i reported - This also works perfectly with 0.13.

And i'm fine with DOSBox Pure displaying the wrong .CUE name as long as the image can be mounted and the game works. DBP shows the .CUE extension anyway so people can still find the image files even if the names are wrong (might be a little confusing with games that use multiple CD's though).

I'll leave this issue open for a couple of days incase anyone else wants to report back on the 0.13 fix.
